### PR TITLE
Depend on python-lsp-server

### PR DESCRIPTION
**Pull request by haplo**
[Palantir's python-language-server](https://github.com/palantir/python-language-server) is now unmaintained. [python-lsp-server](https://github.com/python-lsp/python-lsp-server) is a community-maintained fork.

See https://github.com/tomv564/pyls-mypy/issues/51 for a discussion.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,2 +1,5 @@
-python-language-server>=0.34.0
-mypy
+future;python_version < '3'
+flake8
```
Comment (Richardk2n) [inline requirements.txt:2]: Where do the additional requirements come from?

Comment (haplo) [reply requirements.txt:2]: Sorry I think they came from the old pyls-mypy branch I based my changes on (see https://github.com/tomv564/pyls-mypy/pull/52). I will fix it.

File: mypy_ls/plugin.py [hunk 2]
```diff
@@ -187,10 +189,11 @@ def init(workspace: str) -> Dict[str, str]:
...
     global mypyConfigFile
     mypyConfigFile = findConfigFile(workspace, "mypy.ini")
-    if (("enabled" not in configuration or configuration["enabled"])
-       and ("live_mode" not in configuration or configuration["live_mode"])):
+    if ("enabled" not in configuration or configuration["enabled"]) and (
+        "live_mode" not in configuration or configuration["live_mode"]
+    ):
         global tmpFile
-        tmpFile = tempfile.NamedTemporaryFile('w', delete=False)
+        tmpFile = tempfile.NamedTemporaryFile("w", delete=False)
```
Comment (haplo) [inline mypy_ls/plugin.py:196]: I didn't realize that my editor automatically blackens all Python files, I didn't mean to push these changes. I will revert them.

Comment (Richardk2n) [reply mypy_ls/plugin.py:196]: Some uniform formatting would probably be good and I did a very poor job in that regard. Therefore, this is fine with me.

Comment (haplo) [reply mypy_ls/plugin.py]: Oh well I already pushed a revert. If you are OK with formatting with black I will open an issue and do it properly, with a pre-commit hook probably.

File: README.rst [hunk 3]
```diff
@@ -31,7 +31,7 @@ Depending on your editor, the configuration (found in a file called mypy-ls.cfg
 ::
 
     {
-	"enabled": True,
-	"live_mode": True,
-	"strict": False
+    "enabled": True,
```
Comment (haplo) [inline README.rst:34]: This was my editor replacing the tabs with spaces, sorry about that.

File: mypy_ls/plugin.py (hunk 2 above)
Comment (Richardk2n) [reply mypy_ls/plugin.py]: That would be nice

**Comment by Richardk2n** [comment]
@haplo Do you want to include any other changes in this PR? Otherwise this looks fine.

**Comment by haplo** [comment]
> @haplo Do you want to include any other changes in this PR? Otherwise this looks fine.

I would keep this to the minimum to make it compatible with python-lsp-server. We can open separate issues for other changes.

Feel free to squash merge.

**Pull request merged**